plugin-relay/src/server.rs

```diff
@@ -68,20 +71,38 @@ impl AlumetPlugin for RelayServerPlugin {
     }
 
     fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
-        let config = deserialize_config(config)?;
+        let mut config: Config = deserialize_config(config)?;
+
         Ok(Box::new(RelayServerPlugin { config }))
     }
 
     fn start(&mut self, alumet: &mut AlumetStart) -> anyhow::Result<()> {
-        let addr: SocketAddr = match self.config.ipv4_only {
-            true => SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, self.config.port)),
-            false => SocketAddr::V6(SocketAddrV6::new(
-                Ipv6Addr::LOCALHOST,
+        // Retrieve the ip from the config using
+        // [the trait ToSocketAddrs](https://doc.rust-lang.org/stable/std/net/trait.ToSocketAddrs.html)
+        // of (String, u16) to use a raw address without any port.
+        let current_ip = match (self.config.address.clone(), self.config.port)
+            .to_socket_addrs()
+            .context("Parsing of the `address` param of the server relay plugin")?
+            .next()
+        {
+            Some(socket) => socket.ip(),
+            _ => {
+                return Err(anyhow!("Could not retrieve any ip from {}", self.config.address));
+            }
+        };
+
+        // Once the IP has been parsed correctly, we need to build from scratch
+        // the SocketAddr to allow us to add `ipv6_scope_id` if needed.
+        let addr = match current_ip {
+            IpAddr::V4(ipv4) => SocketAddr::V4(SocketAddrV4::new(ipv4, self.config.port)),
```

Comment:
If config.port is != socket.port() you could print a warning to explain that the port is ignored because it's in a separate config field, what do you think?

Reply:
Well, when using to_socket_addrs on (String, u16), the string cannot have a port specified. It will raise the following error:

```
 Caused by:
        0: Parsing of the `address` param of the server relay plugin: 127.0.0.1:50051
        1: failed to lookup address information: Name or service not known
```

[See](https://doc.rust-lang.org/stable/std/net/trait.ToSocketAddrs.html)